xlators/system/posix-acl/src/posix-acl.c

```diff
@@ -468,71 +480,53 @@ __posix_acl_set_specific(inode_t *inode, xlator_t *this, gf_boolean_t is_access,
     return ret;
 }
 
-int
+static void
 __posix_acl_set(inode_t *inode, xlator_t *this, struct posix_acl *acl_access,
                 struct posix_acl *acl_default)
 {
-    int ret = 0;
     struct posix_acl_ctx *ctx = NULL;
 
     ctx = posix_acl_ctx_get(inode, this);
     if (!ctx)
-        goto out;
+        return;
```

Comment:
Probably not returning -1 here was a bug.

Reply:
This is even a bit stranger - it can only fail if posix_acl_ctx_get() failed, now it is unlikely to fail since it was called right after __posix_acl_get() which also did posix_acl_ctx_get() (and as exaplained earlier, the return code was not checked, but the variables it provided were.

So perhaps a better fix would be:
```
if (acl_access || acl_default)
    __posix_acl_set(inode, this, acl_access, acl_default);
```

(and before all that, check that inode is not NULL, which in theory can fail all of the above) ?

Reply:
I'm not sure if we should consider that an error accessing the context is the same as the presence or not of an ACL. Not having an ACL may be ok, but not being able to access it should be an error, because if it's really there but we can't access it, we can't assume that it's NULL (it could allow unwanted accesses).

However I'm not sure if the logic of the xlator allows a ctx with `acl_access` and/or `acl_default` as NULL. If it's not allowed, then not checking the returned code is fine and we could just check the pointer, but if we are sure the pointer won't be NULL in the context, why not just check the error code, which seem simpler, and assume the pointer is not NULL if there's no error ?

If one of the pointers can be NULL, and the caller just wants this pointer, how we do differentiate between an empty ACL and an error accessing it ?

Reply:
Looks like it can happen only on some error conditions. For example, multiple different failure scenarios in posix_acl_from_xattr() can give you a NULL ACL.